src/main/java/mate/academy/hibernate/relations/dao/impl/MovieDaoImpl.java

```diff
@@ -12,11 +15,38 @@ public MovieDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Movie add(Movie movie) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(movie);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add the movie: " + movie, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return movie;
```

Comment:
According to the checklist, you don't need to set an ID for your entity explicitly as it will be done by Hibernate. Ensure that the `Movie` entity is correctly annotated so that Hibernate manages the ID generation.